src/html/to_html_tree.ml

```diff
@@ -811,12 +811,26 @@ struct
           }
 
       | `Nested_article item ->
-        let html, subpages = state.render_nested_article item in
+        let html, maybe_docs, subpages = state.render_nested_article item in
         let html, maybe_id = add_anchor state.item_to_id item html in
         let a = add_spec state.item_to_spec item maybe_id in
+        let html =
+          match maybe_docs with
+          | [] -> Html.article ~a html
+          | docs ->
+            let docs = Documentation.first_to_html docs in
+            let docs = (docs :> (Html_types.div_content Html.elt) list) in
+            let adoc =
+              [ Html.a_class ["doc"]
+              ; Html.a_style "padding: 0.75em; padding_bottom: 0;" ]
+            in
+            Html.article
```

Comment:
I'm not sure why this needs to be an `article`. Other signature items are generated as `dl/dt/dd` tags, I think modules shouldn't be an exception.

Reply:
I think the original motivation for `<article>` (or something other than `<dt>`) was that for inlined modules and classes, we will have the content of almost an entire separate page inside the tag. For other kinds of signature items, the item goes in `<dt>` and the docs in `<dd>`. However, the docs for an inline module don't follow the whole module. They are nested inside the module, immediately after the module name.

Reply:
> I think the original motivation for <article> (or something other than <dt>) was that for inlined modules and classes, we will have the content of almost an entire separate page inside the tag. 

I think the non-`dl` tag only makes sense for _actually_ inlined modules (do we have any plans for this, btw?). For the cases where the module is not inlined, and only has the brief, I'd still use a description list.

Reply:
That was the plan, but I was originally going to inline all modules by default last year, so `<article>` was in preparation for that. I agree with undoing it for the time being, if anyone wants to.

Reply:
I'd be happy to do it! :)

---

Can you elaborate on the trade-offs between inlining/referencing nested items?

In some cases referencing creates an additional indirection for the users. In larger libraries having all modules inlined is too verbose. Would be very useful to allow users to pass inline hints for odoc. Additionally the inlined modules could be collapsable similar to current includes.

Reply:
These are the same tradeoffs I had in mind.

The basic argument for inlining by default was that if it was inline in the `.mli`, and looked fine, it will probably look fine if inlined in the HTML.

Reply:
@dbuenzli I’m curious to know what’s your opinion on this, given that most of your libraries tend to expose one single interface.

Would you be happier with inlined modules, the same way they are defined in interface files?

Reply:
No I would not be happy at all. I think I had this discussion with almost every person that worked on the markup after all these years and some earlier version actually did it and it made the docs of my libraries unreadable. Then usually it would be proposed to have them in a details element, but then I would point out this would break anchor linking, usually it would then be proposed that this should be controlled via an annotation -- which is fine to me as long as the default is kept the way `ocamldoc` works. The latter is the argument to have here which is to follow the way `ocamldoc` has worked for all these years since most people who care about documentation actually designed their documentation interaction around its output structure.

Reply:
@dbuenzli I'm glad I asked then. As you can imagine I wasn't aware of any previous discussions. Feel free to point me to any relevant discussions you think I should be aware of.

I'm fine with keeping ocamldoc's convention here as the default behaviour.